### PR TITLE
GUAC-1244: Added timeout to mouse scroll events to catch and issue mi…

### PR DESCRIPTION
**Pull request by flangelo**
Small scroll events are not caught and issued to remote if the scroll threshold is not reached.
Added timeout to mouse scroll events to catch and issue minor scrolling to remote.

### CODE REVIEW

File: guacamole-common-js/src/main/webapp/modules/Mouse.js [hunk 1]
```diff
@@ -40,6 +40,12 @@ Guacamole.Mouse = function(element) {
     var guac_mouse = this;
 
     /**
+     * Timeout before a single mouse scroll happens.
+     * @type type
```
Comment (mike-jumper) [inline guacamole-common-js/src/main/webapp/modules/Mouse.js]: `@type Number` perhaps? `window.setTimeout()` returns a numerical ID.

File: guacamole-common-js/src/main/webapp/modules/Mouse.js [hunk 2]
```diff
@@ -273,38 +279,46 @@ Guacamole.Mouse = function(element) {
...
+                            guac_mouse.onmousedown(guac_mouse.currentState);
+                        }
+
+                        if (guac_mouse.onmouseup) {
+                            guac_mouse.currentState.up = false;
+                            guac_mouse.onmouseup(guac_mouse.currentState);
+                        }
+
+                        scroll_delta += guac_mouse.scrollThreshold;
+                                } while (scroll_delta < 0);
```
Comment (mike-jumper) [inline guacamole-common-js/src/main/webapp/modules/Mouse.js]: Please follow the style of the surrounding code. It looks like this is partially indented with tabs, and partially indented with spaces, leading to brain-melting misalignment.

We use a four-space indent across the board - no tabs please.

File: guacamole-common-js/src/main/webapp/modules/Mouse.js (hunk 1 above)
Comment (flangelo) [inline guacamole-common-js/src/main/webapp/modules/Mouse.js]: Yes, Number is good. Missed that one.

**Comment by flangelo** [comment]
Fixed.

**Comment by mike-jumper** [comment]
Thanks for fixing all this, @flangelo. After testing and reviewing these changes, I've found the following issues:

1. 100 millisecond latency between the mouse scroll wheel click and response via Guacamole.
2. Over-emphasis of small scroll events for devices which use true smooth scrolling, and actually provide real pixel values in `e.delta`.

I think, generally, the use of a timer is not fundamentally necessary, and we should be able to fix the problem described by [GUAC-1244](https://glyptodon.org/jira/browse/GUAC-1244) by using better constants and resetting `scroll_delta`, as you have done here.

I have created #209 which includes and builds off these changes.

Merging through #209.

**Pull request closed without merging**